Adds ModalConfirmDelete.

webinista opened the pull request:
Closes department-of-veterans-affairs/messaging-fe#48
- Added reducers/modals.js, actions/modals.js
- Removed modal functionality from actions/compose.js and reducers/compose.js.
- Made some modal props required.
- Added ability to set an additional class name for a modal.

webinista committed:
Adds ModalConfirmDelete.

Closes department-of-veterans-affairs/messaging-fe#48

- Added reducers/modals.js, actions/modals.js
- Removed modal functionality from actions/compose.js and reducers/compose.js.
- Made some modal props required.
- Added ability to set an additional class name for a modal.

U-DON reviewed (changes requested):
Mostly LGTM with some nits and questions.

webinista commented on src/js/messaging/components/Modal.jsx:
```diff
@@ -1,10 +1,35 @@
+  }
+
   render() {
+    const modalCss = classNames('rx-modal', this.props.cssClass);
```
Yes. The plan is to make it a common component.

webinista commented on src/js/messaging/containers/Compose.jsx:
```diff
@@ -43,71 +51,88 @@ class Compose extends React.Component {
+  handleDelete(domEvent) {
+    // TODO: Dispatch an action that makes this API call
+    domEvent.preventDefault();
+    browserHistory.push(paths.BASE_URL);
```
Actually, according to the prototype, it should go back to the drafts view not the Inbox. I need to fix that.

webinista committed:
Merge branch 'master' into department-of-veterans-affairs/messaging-fe#48

webinista commented on src/js/messaging/containers/Compose.jsx:
```diff
@@ -24,13 +26,19 @@ import {
     this.handleSubjectChange = this.handleSubjectChange.bind(this);
-    this.handleMessageChange = this.handleMessageChange.bind(this);
+    this.handleDeleteDraft = this.handleDeleteDraft.bind(this);
+    this.handleDelete = this.handleDelete.bind(this);
```
`handleDeleteDraft` triggers the confirmation modal. `handleDelete` handles the actual deletion. Removed `handleDeleteDraft` in favor of using `toggleDeleteConfirm` in both places.

webinista committed:
Improved method names, path constant names.

- Removed handleDeleteConfirm.
- Renamed handleDelete to handleConfirmDelete
- Changed paths constants names.

webinista commented:
Is this okay to merge? (If so, please do the honors.)
